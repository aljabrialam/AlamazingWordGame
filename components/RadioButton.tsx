import React, {useState} from 'react';
import {Text, TouchableOpacity, View} from 'react-native';

const RadioButton = (props: any) => {
  const [radio, setRadio] = useState('');

  const select = (value: any) => {
    setRadio(value);
    props.onSelect(value);
  };

  return (
    <View>
      {props.data.map((q: any) => {
        return (
          <View
            style={{
              backgroundColor: radio === q.value ? '#D9E7FB' : '#fff',
              borderColor: '#d3d6da',
              borderWidth: 2,
              borderRadius: 10,
              height: 50,
              width: 240,
              alignItems: 'center',
              paddingStart: 30,
              paddingEnd: 30,
              marginTop: 10,
            }}
            key={q.label}>
            <TouchableOpacity
              hitSlop={{top: 30, bottom: 30, left: 30, right: 30}}
              style={{
                flexDirection: 'row',
              }}
              onPress={() => select(q.value)}>
              <Text
                style={{
                  //backgroundColor: radio === q.value ? '#d3d6da' : '#fff',
                  marginTop: 8,
                  fontSize: props.styling.size - 10,
                }}>
                {q.label}
              </Text>
            </TouchableOpacity>
          </View>
        );
      })}
    </View>
  );
};

export default RadioButton;
