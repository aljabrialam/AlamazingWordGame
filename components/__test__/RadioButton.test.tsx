import React from 'react';
import renderer from 'react-test-renderer';

import RadioButton from '../RadioButton';

const navigationPropsLanding = {
  navigation: {
    navigate: jest.fn(),
    addListener: jest.fn()
  },
  route: {params: {radio: 'Cities'}}
}

const data = [
    {
      label: 'Cities',
      value: 'Cities'
    },
    {
      label: 'Foods',
      value: 'Foods'
    },
    {
      label: 'Animals',
      value: 'Animals'
    }
  ]

describe('<RadioButton />', () => {
  it('has 1 child', () => {

    const tree: any = renderer.create(<RadioButton data={data} styling={{ color: 'blue', size: 30 }} {...navigationPropsLanding} />).toJSON();
    expect(tree.children.length).toBe(3);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<RadioButton data={data} styling={{ color: 'blue', size: 30 }} {...navigationPropsLanding} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});