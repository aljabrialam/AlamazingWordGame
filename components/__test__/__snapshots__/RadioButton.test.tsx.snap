// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> renders correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#d3d6da",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 50,
        "marginTop": 10,
        "paddingEnd": 30,
        "paddingStart": 30,
        "width": 240,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "marginTop": 8,
          }
        }
      >
        Cities
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#d3d6da",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 50,
        "marginTop": 10,
        "paddingEnd": 30,
        "paddingStart": 30,
        "width": 240,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "marginTop": 8,
          }
        }
      >
        Foods
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#d3d6da",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 50,
        "marginTop": 10,
        "paddingEnd": 30,
        "paddingStart": 30,
        "width": 240,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "marginTop": 8,
          }
        }
      >
        Animals
      </Text>
    </View>
  </View>
</View>
`;
